### PR TITLE
Some Azure updates

### PR DESCRIPTION
**Pull request by TheRealFalcon**
- Add ability to make more granular changes to vm_params when creating
an instance
- When deprovisioning an image for snapshot generation,
add option to keep the original user instead of automatically
removing it
- Small bugfix to not throw exception if image offer is None

### CODE REVIEW

File: pycloudlib/azure/cloud.py [hunk 1]
```diff
@@ -376,7 +376,7 @@ def _create_vm_parameters(self, name, image_id, nic_id, user_data):
         return vm_parameters
 
     def _create_virtual_machine(
-            self, image_id, nic_id, user_data, name, **kwargs
+        self, image_id, nic_id, user_data, name, vm_params=None
```
Comment (TheRealFalcon) [inline pycloudlib/azure/cloud.py:379]: @blackboxsw or @lucasmoura Would this break anything in ua-client? Are you guys overriding Azure vm_params via here (and line 404) anywhere?

Comment (blackboxsw) [reply pycloudlib/azure/cloud.py:379]: Safe from uaclient standpoint, integration tests  only currently call launch with image_id and user_data params and snapshot(instance=X).

File: pycloudlib/azure/util.py [hunk 2]
```diff
@@ -179,7 +179,7 @@ def is_pro_image(image_id, registered_image):
     if img_dict.get("publisher") == "Canonical":
         offer = img_dict["offer"]
     elif registered_image is not None:
-        offer = registered_image["offer"]
+        offer = registered_image["offer"] or ""
```
Comment (blackboxsw) [inline pycloudlib/azure/util.py:182]: Any reason we pefer this syntax to registered_image.get("offer", "") ?

Comment (TheRealFalcon) [reply pycloudlib/azure/util.py:182]: Yeah, I tried that syntax, but `offer` actually exists as a key, but the value is `None`. The return statement fails trying to iterate across None.

**Review by blackboxsw — APPROVED** [review]
LGTM!

**Pull request merged**